src/HBL01MItineraryHashtag/HBL01MItineraryHashtag.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Client } from '@elastic/elasticsearch';
+import { firestore } from 'firebase-admin';
+import { DocumentData, QueryDocumentSnapshot } from 'firebase-admin/firestore';
+
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+
+import { Itineraries } from 'spelieve-common/lib/Models/Itinerary/IDB01/Itineraries';
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+// import { PBL002UpsertPlaceDataServiceRule } from './PBL002UpsertPlaceData.rule';
+
+@Injectable()
+export class HBL01MItineraryHashtagService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async doExecute() {
+    // リクエストに合致する QueryDocumentSnapshot を取得する
+    const itineraryCollectionRef = firestore().collection(
+      Itineraries.modelName,
+    );
+    const itineraryDocumentSnap: QueryDocumentSnapshot<DocumentData>[] | [] =
+      await itineraryCollectionRef
+        .get()
+        .then((qss) => (qss.empty ? [] : qss.docs));
+
+    const tag_list: Array<string> = itineraryDocumentSnap
+      .map((doc) => doc.data().tags)
+      .flat();
+
+    // count tags frequency
+    const toDict = (arr: Array<string>): { [key: string]: number } => {
+      const dict: { [key: string]: number } = {};
+      arr.map((e) => {
+        if (dict[e]) {
+          undefined += 1;
+        } else {
+          dict[e] = 1;
+        }
+      });
+      return dict;
+    };
+    const tagsDict: { [key: string]: number } = toDict(tag_list);
```

Comment:
Array<string>　ではなく、MItineraryHashtag[]にしたい

実装は↓みたいな感じ。
spelieve-commonがもしかしたら更新されていないから更新しないとバグるかも

後、created_atが漏れてたから追加してほしい（typescript で気づけた！）

```suggestion
    const mItineraryHashTagList: m_itinerary_hashtags[] = []
    const created_at = new Date();
    itineraryDocumentSnap.docs.map(doc => doc.data())
    .forEach((itinerary) => {
      itinerary.tags.forEach(tag => {
        const element = mItineraryHashTagList.find(value => value.tag === tag);
        if (element) {
            element.attached_count += 1;
        } else {
          mItineraryHashTagList.push({
              attached_count: 1,
              tag: tag,
              created_at: created_at
            });
        }
      }
      )
    });
```

Reply:
[2d4af66](https://github.com/Ayato-kosaka/spelieve-backend-tag/pull/3/commits/2d4af66d0d35dd9a88717d4c0103967afd9895b0)